Reopening pull request #2065 forum userlist wrapping

rica-carv opened the pull request:
Reopening issue #2065 

### Types of Changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist
- [x] My code adheres to the e107 [code style standard](https://github.com/e107inc/e107/wiki/e107-Coding-Standard).
- [ ] I have read the [**contributing** document](https://github.com/e107inc/e107/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/e107inc/e107/tree/master/e107_tests) to cover my changes.
- [ ] All new and existing tests pass.

rica-carv committed:
Forum userlist shortcode wrapping

Removed hardcoded html and use wrapping instead, for userlist shortcode...

rica-carv committed:
Update forum_template.php

Jimmi08 commented:
@rica-carv 
I appreciate your work so now I am trying to understand what you did. 

This PR will be never approved.  forum shortcodes file is already updated and there is a conflict in it with your code.  

In the original PR @CaMer0n already told you that it breaks old themes.  My objections are that you didn't fix this shortcode fully.

Link to user profile should use e107::getUrl() way, not hardcoded HTML code. On one place you are removing it, but on other place you let it there.  

global $listuserson; - should be `$listuserson = e107::getOnline()->userList(); `

return;  should be `return ""; `

user link should be:
```

		$uparams = array('id' => $oid, 'name' => $oname;
		$link = e107::getUrl()->create('user/profile/view', $uparams);
```

Everything is already managed (new way) in online plugin 
It should be done similar way as in $ONLINE_MENU_TEMPLATE['default']['online_members_list_extended']  

I hope it helps somehow.  The original issue should be opened, because this should be fixed. Thanks

rica-carv commented:
@Jimmi08 Sorry, but i don't understand your post. If you check both files i changed, i only take out hardcode html inside the forum_shortcodes.php file to forum_template.php file!
I didn't do anything else...

All the changes you're speaking were already there, so i don't see any other conflict....

I think you're mixing up old change with my new ones.... and i don't understand the changes you're talking about....

rica-carv committed:
Add files via upload

rica-carv committed:
Add files via upload

rica-carv committed:
Merge branch 'master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

rica-carv committed:
Merge branch 'e107inc:master' into rica-carv-forum_userlist_wrapping

e107help committed:
fix(forum): keep the USERLIST label when a theme supplies its own template

The wrapper was assigned inside the `if (!isset($FORUM_MAIN_END))` guard, so
a theme that defines its own $FORUM_MAIN_END skipped the whole block and lost
the "who is online" label and link that sc_userlist() used to concatenate.
This is the compatibility break CaMer0n raised on #2065 in 2016.

Assign it above the guard instead, and only when the theme has not already
set it. Restore e_HTTP on the online.php link to match the sibling links in
this file, and tidy the indentation left behind by the extracted strings.

Deltik commented on e107_plugins/forum/templates/forum_template.php:
```diff
@@ -29,14 +29,23 @@
 */
 // LEGACY definition with LAN Shortcodes ({INFOTITLE}).....
-$FORUM_MAIN_END = "</table></div>\n<div class='spacer'>\n<table style='".USER_WIDTH."' class='fborder table'>\n<tr>\n<td colspan='2' style='width:60%' class='fcaption'>{INFOTITLE}</td>\n</tr>\n<tr>\n<td rowspan='4' style='width:5%; text-align:center' class='forumheader3'>{LOGO}</td>\n<td style='width:auto' class='forumheader3'>{USERINFO}</td>\n</tr>\n<tr>\n<td style='width:95%' class='forumheader ...
+$FORUM_MAIN_END ="</table></div>\n<div class='spacer'>\n<table style='".USER_WIDTH."' class='fborder table'>\n<tr>\n<td colspan='2' style='width:60%' class='fcaption'>{INFOTITLE}</td>\n</tr>\n<tr>\n<td rowspan='4' style='width:5%; text-align:center' class='forumheader3'>{LOGO}</td>\n<td style='width:auto' class='forumheader3'>{USERINFO}</td>\n</tr>\n<tr>\n<td style='width:95%' class='forumheader3 ...
```
@e107help, you somehow removed a space here after the assignment equal sign:

```suggestion
$FORUM_MAIN_END = "</table></div>\n<div class='spacer'>\n<table style='".USER_WIDTH."' class='fborder table'>\n<tr>\n<td colspan='2' style='width:60%' class='fcaption'>{INFOTITLE}</td>\n</tr>\n<tr>\n<td rowspan='4' style='width:5%; text-align:center' class='forumheader3'>{LOGO}</td>\n<td style='width:auto' class='forumheader3'>{USERINFO}</td>\n</tr>\n<tr>\n<td style='width:95%' class='forumheader3'>{INFO}</td>\n</tr><tr>\n<td style='width:95%' class='forumheader3'>{FORUMINFO}</td>\n</tr>\n<tr>\n<td style='width:95%' class='forumheader3'>{USERLIST}<br />{STATLINK}</td>\n</tr>\n</table>\n</div>\n<div class='spacer'>\n<table class='fborder table' style='".USER_WIDTH."'>\n<tr>\n<td class='forumheader3' style='text-align:center; width:33%'>{ICONKEY}</td>\n<td style='text-align:center; width:33%' class='forumheader3'>{SEARCH}</td>\n<td style='width:33%; text-align:center; vertical-align:middle' class='forumheader3'><span class='smallblacktext'>{PERMS}</span>\n</td>\n</tr>\n</table>\n</div>\n</div>";
```

You should amend your commit to tighten the formatting.
